truss/templates/control/control/endpoints.py

```diff
@@ -105,9 +112,16 @@ def stop_inference_server(request: Request) -> Dict[str, str]:
     return {"msg": "Inference server stopped successfully"}
 
 
+def _decode_body(resp):
+    if "application/octet-stream" in resp.headers["content-type"]:
+        return truss_msgpack_deserialize(resp.content)
+    return resp.json()
+
+
 def _is_model_not_ready(resp) -> bool:
-    return (
-        resp.status_code == 503
-        and resp.content is not None
-        and "model is not ready" in resp.content.decode("utf-8")
-    )
+    if resp.status_code != 503:
```

Comment:
I think this was a little bit easier to parse/read when it was a single boolean expression, this requires a little extra reading:

```
    return (
        resp.status_code == 503
        and resp.content is not None
        and "model is not ready" in_decode_body(resp)
    )
```